src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/fenced_linux.py

```diff
@@ -40,6 +49,75 @@ def start(self, force=False):
 
         return proc.returncode
 
-    def stop(self):
+    def stop(self, banhammer=True):
+        if banhammer:
+            # dont care if it's running or not just
+            # SIGKILL anything that has "fenced" in
+            # the process name
+            subprocess.run(['pkill', '-9', '-f', 'fenced'])
+        else:
+            res = self.middleware.call_sync('failover.fenced.run_info')
+            if res['running'] and res['pid']:
+                os.kill(res['pid'], signal.SIGKILL)
+
+    def run_info(self):
+        res = {'running': False, 'pid': ''}
+        with contextlib.suppress(Exception):
+            with open(PID_FILE, 'rb') as f:
+                res['pid'] = int(f.read().decode())
+
+        check_running_procs = False
+        if res['pid']:
+            try:
+                os.kill(res['pid'], IS_ALIVE_SIGNAL)
+            except OSError:
+                check_running_procs = True
+            else:
+                res['running'] = True
+        else:
+            check_running_procs = True
+
+        if check_running_procs:
+            # either 1. no pid in file or 2. pid in file is wrong/stale
+            _iter = psutil.process_iter
+            proc = 'fenced'
+            res['pid'] = next((p.pid for p in _iter() if p.name() == proc), '')
+            res['running'] = bool(res['pid'])
+
+        return res
+
+    def signal(self, options):
+        res = self.middleware.call_sync('failover.fenced.run_info')
+        if res['running'] and res['pid']:
+            try:
+                if options.get('reload', False):
+                    os.kill(res['pid'], signal.SIGHUP)
+                if options.get('log_info', False):
+                    os.kill(res['pid'], signal.SIGUSR1)
+            except OSError as e:
+                raise CallError(f'Failed to signal fenced: {e}')
+
+
+async def hook_pool_event(middleware, *args, **kwargs):
```

Comment:
Just something to consider, its just a matter of personal preference, so no need to change, but for myself I like to return first to avoid deep indentation levels, e.g.:

```
# We only care for SCALE Enterprise
if await middleware.call('system.product_type') != 'SCALE_ENTERPRISE':
    return
    
# HA Licensed handle fenced on their own, we only want mixed Controllers
 if await middleware.call('failover.licensed'):
     return

...
```

Reply:
Yeah, I like that too. I've changed it.